nemo_text_processing/inverse_text_normalization/hi/taggers/decimal.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2024 and onwards Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.import pynini
+
+import pynini
+from pynini.lib import pynutil
+
+from nemo_text_processing.inverse_text_normalization.hi.utils import get_abs_path
+from nemo_text_processing.text_normalization.en.graph_utils import (
+    INPUT_CASED,
+    INPUT_LOWER_CASED,
+    MIN_NEG_WEIGHT,
+    MINUS,
+    NEMO_DIGIT,
+    NEMO_SIGMA,
+    TO_LOWER,
+    GraphFst,
+    capitalized_input_graph,
+    delete_extra_space,
+    delete_space,
+)
+from nemo_text_processing.text_normalization.en.utils import load_labels
+
+
+def get_quantity(
+    decimal: 'pynini.FstLike', cardinal_up_to_hundred: 'pynini.FstLike', input_case: str = INPUT_LOWER_CASED
+) -> 'pynini.FstLike':
+    """
+    Returns FST that transforms either a cardinal or decimal followed by a quantity into a numeral,
+    e.g. दस लाख -> integer_part: "१॰" quantity: "लाख"
+    e.g. एक दशमलव पाँच लाख -> integer_part: "१" fractional_part: "५" quantity: "लाख"
+ 
+    Args: 
+        decimal: decimal FST
+        cardinal_up_to_hundred: cardinal FST
+        input_case: accepting either "lower_cased" or "cased" input.
+    """
+    numbers = cardinal_up_to_hundred @ (
+        pynutil.delete(pynini.closure("0")) + pynini.difference(NEMO_DIGIT, "0") + pynini.closure(NEMO_DIGIT)
+    )
+    suffix = pynini.union("हजार", "लाख", "करोड़", "अरब", "खरब", "नील", "पद्म", "शंख",)
```

Comment:
can we add these as a data file and read from it?

Reply:
Done